flask_admin/model/base.py

```diff
@@ -1477,6 +1477,10 @@ def create_view(self):
             return redirect(return_url)
 
         form = self.create_form()
+        if self._form_create_rules:
```

Comment:
It is slightly different. In method that calls `edit_form` and getting form class, it can call something like `self._validate_form_class` (or whatever name could be). `_validate_form_class` will call `self._check_render_rules` twice with default implementation that'll filter invisible fields out of the forms. 

Here's rationale:
1. Form can be supplied by the developer, without using Flask-Admin scaffolding. It might contain custom fields that contribute to the logic of the application without changing model itself (you can do all kind of stuff in the `on_model_change`, as form object is passed there). These fields should not be filtered out automatically.

2. Keeping previous point in mind, it should be easy to change default behavior. I'm not yet convinced that straight form class filtering is right default behavior. But agree that at least it should issue a warning. 

3. If developer wants to filter form using list of visible fields - he should be able to do it from any part of the code. In lots of cases there's no need to filter form in views - it won't change in most of the cases. If it is necessary to filter form dynamically, it is always possible to do it in `get_form` or create/edit overloads.

Hope it all makes sense.